wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+use CRM_Pets_ExtensionUtil as E;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+	
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+	
+		// Check if the 'Eck_Collection_Camp1' exists
+		$collectionCampData = $decodedData['Eck_Collection_Camp1'][0]['fields'] ?? null;
+		if ($collectionCampData === null) {
+			return;
+		}
```

Comment:
This should be good enough I think
```suggestion
		if (!array_key_exists('Eck_Collection_Camp1', $decodedData)) {
			return;
		}
```